src/expression/node/SymbolNode.js

```diff
@@ -68,16 +69,16 @@ export const createSymbolNode = /* #__PURE__ */ factory(name, dependencies, ({ m
       }
     } else if (name in math) {
       return function (scope, args, context) {
-        return name in scope
-          ? getSafeProperty(scope, name)
+        return hasMapProperty(scope, name)
+          ? getMapProperty(scope, name)
           : getSafeProperty(math, name)
       }
     } else {
       const isUnit = isValuelessUnit(name)
 
       return function (scope, args, context) {
-        return name in scope
-          ? getSafeProperty(scope, name)
+        return hasMapProperty(scope, name)
+          ? getMapProperty(scope, name)
           : isUnit
             ? new Unit(null, name)
```

Comment:
Once `UnitMath` is operational, I'd like to consider moving unit lookup into scope, rather than directly from the AST.

Reply:
I think with the new `Map` interface approach, we can indeed implement looking up whether something is a unit with a third read-only "scope", I guess that's what you mean? That's really nice!

Reply:
I haven't thought this through massively: potentially, we've got multiple scopes, and ways to combine them: e.g. 

Simple scopes, which differ by how symbols and their values are stored:

 * `math` scope, (currently just `math`)
 * `unit` scope (currently looking up valueless units in `Unit`)
 * user defined `variable` and `function` (currently `scope`)

They can be composed in funky ways:

 * Composite Scope: resolves a symbol by looking in multiple child scopes until one is found.
 * Importing Scope: resolves a symbol by looking in itself, then a number of 'imported' scopes.
 * Sub Scope: resolves a symbol by looking in itself, then its parent.

The AST doesn't have to care where the unit, function or variable comes from, just that it comes from `scope.get(name)`.

Reply:
Yes, I have exactly the same idea 😄 . With the `Map` interface this becomes quite trivial (especially hooking up units as a "scope"). I'm not sure but maybe a util function like `getProperty(scope1, scope2, scope3)` could become useful to replace the current conditionals. Maybe that's overkill though, there is not that many places where it is needed I think.